### PR TITLE
Disaster recovery feature

### PR DESCRIPTION
**Pull request by marsaud**

### CODE REVIEW

File: src/xo.js [hunk 1]
```diff
@@ -796,6 +796,41 @@ export default class Xo extends EventEmitter {
...
+        if (vm && reg.test(vm.name_label)) {
+          vms.push(vm)
+        }
+      }
+    })
+    const olderCopies = sortBy(vms)
+
+    const sourceXapi = this.getXAPI(vm)
+    const targetXapi = this.getXAPI(sr)
+    const drCopy = await sourceXapi.remoteCopyVm(vm, targetXapi, sr)
```
Comment (julien-f) [inline src/xo.js]: If I am not mistaken, it should be `vm.id` and `sr.id`.

File: src/xo.js [hunk 2]
```diff
@@ -796,6 +796,41 @@ export default class Xo extends EventEmitter {
...
+
+    const date = safeDateFormat(new Date())
+    const copyName = `${date}_${tag}_${vm.name_label}`
+
+    await targetXapi.call('VM.set_name_label', drCopy.ref, copyName)
```
Comment (julien-f) [inline src/xo.js]: Avoid calling directly the XAPI, implement a basic `Xapi#setVmProperties()` instead.

Comment (julien-f) [inline src/xo.js]: Otherwise, I can implement this functionality directly in `Xapi#remoteCopyVm()`.

Comment (marsaud) [inline src/xo.js]: Why not, but I saw a heavy use of this in vm.coffee ... I would prefer take time specifically on a clean setVmProperties with a global API code review, than doing something basic on the fly for the DR feature...but I'll follow your final word.

File: src/xo.js [hunk 3]
```diff
@@ -796,6 +796,41 @@ export default class Xo extends EventEmitter {
...
+    await targetXapi.addTag(drCopy.id, 'Disaster Recovery')
```
Comment (julien-f) [inline src/xo.js]: I now it's far too messy now, but `drCopy` does is a `xen-api` object (not a XO object`), consequently it does not have an `id` property, but it does have `$id`.

File: src/xo.js (hunk 2 above)
Comment (julien-f) [inline src/xo.js]: It's heavily used because it's legacy code, it should be updated when possible.

Comment (marsaud) [inline src/xo.js]: Yeah, I know, I just suggest we keep this for a specific update effort ... unless you fear the effort would come late ...

**Comment by marsaud** [comment]
To reopen targetting next-release

**Pull request closed without merging**